zerver/views/__init__.py

```diff
@@ -1881,6 +1881,12 @@ def json_fetch_api_key(request, user_profile, password=REQ(default='')):
         return json_error("Your username or password is incorrect.")
     return json_success({"api_key": user_profile.api_key})
 
+@csrf_exempt
+def api_fetch_google_client_id(request):
+    if not settings.GOOGLE_CLIENT_ID:
+        return json_error("GOOGLE_CLIENT_ID is not configured", status=400)
```

Comment:
I strongly prefer status codes to carry meaning - 200 seems wrong because the request cannot be fulfilled, and is by definition not 'OK'. You can make a case for 404 but the 400 indicates that the server understood the request but was not able to fulfill it - older servers with the endpoint missing entirely will 404 and you could identify them by status alone.